### PR TITLE
For discussion - cleanup pwmInit function by reduced use of continue

### PR DESCRIPTION
**Pull request by martinbudden**
This is just to illustrate why I think `continue` is overused in this function.

This is just a partial cleanup to illustrate the point.

### CODE REVIEW

File: src/main/drivers/pwm_mapping.c [hunk 1]
```diff
@@ -643,133 +643,141 @@ pwmIOConfiguration_t *pwmGetOutputConfiguration(void){
...
 
-        if (type == MAP_TO_PWM_INPUT && !init->useParallelPWM)
-            continue;
+    if (type == MAP_TO_PWM_INPUT && !init->useParallelPWM)
+        return true;
+
+    if (type == MAP_TO_PPM_INPUT && !init->usePPM)
+        return true;
+
+    return true;
```
Comment (ledvinap) [inline src/main/drivers/pwm_mapping.c]: `false` here ...

Comment (martinbudden) [inline src/main/drivers/pwm_mapping.c]: Oops! Well spotted.

**Comment by hydra** [comment]
yes, extracting the logic that way is better for sure.  can you rebase and complete this PR?

File: src/main/drivers/pwm_mapping.c [hunk 2]
```diff
@@ -643,133 +645,145 @@ pwmIOConfiguration_t *pwmGetOutputConfiguration(void){
...
+
+    setup = hardwareMaps[i];
 
-        if (type == MAP_TO_PPM_INPUT && !init->usePPM)
+    for (i = 0; i < USABLE_TIMER_CHANNEL_COUNT && setup[i] != 0xFFFF; i++) {
+        const uint8_t timerIndex = setup[i] & 0x00FF;
+        uint8_t type = (setup[i] & 0xFF00) >> 8;
+        const timerHardware_t *timerHardwarePtr = &timerHardware[timerIndex];
+
+        if (pwmIsUsed(init, type, timerHardwarePtr, timerIndex)) {
```
Comment (digitalentity) [inline src/main/drivers/pwm_mapping.c]: Not sure that `pwmIsUsed` is a good name, it doesn't tell that it's actually a check if we can use a given pwm output/input. Maybe a better name is `pwmChannelCanBeUsed` ?

**Comment by martinbudden** [comment]
Rebased and @ledvinap 's comment incorporated.

This PR was only rely as a discussion point to show my aversion to the use of the `continue` statement. But you are welcome to merge it in if you like.

It is untested, since I'm not at home at the moment and don't have any hardware for testing, so it should be tested before a merge.

**Comment by hydra** [comment]
@martinbudden thanks for the rebase.   yes, it's very time consuming to test that code.  i really hate the PWM mapping code.  I would like to make it unit testable so we can sure the pwm mapping code really does work as it's a constant source of issues.

**Comment by hydra** [comment]
I think the loop code in the loop should be extracted too.  something like `io = pwmAllocate(usage)`

perhaps also `pwmChannelCanBeUsed` should be `pwmIsChannelAllocatable`?

**Comment by martinbudden** [comment]
@hydra , I think it's probably not worth spending too much time in this area. I believe there are plans afoot to revamp it all for betaflight 3.1 as part of the user-assignable IO. I think @blckmn is the person working on this. Once Cleanflight is converted to the new IO, you'll be able to pick this up for free.

**Comment by ledvinap** [comment]
@hydra: Using RESOURCE from IO (and timer eventually) will help a bit.
Pin allocation order is in general different from pin function priority, so at least nonstandard cases has to be handled explicitly.
IMO it is best to use user-supplied mapping and only check for collisions on FC

**Comment by hydra** [comment]
@martinbudden ok i will close this now then.

**Pull request closed without merging**